onert-micro/onert-micro/include/onert-micro-train.h

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#ifndef ONERT-MICRO-TRAIN_H_
+#define ONERT-MICRO-TRAIN_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/*
+ * typical training flow in onert-micro
+ *
+ * 1. load model or checkpoint
+ *   1-1. (optional) configure training options
+ * 2. feed training input / output(e.g. label) data (cf. unit of a step)
+ * 3. train a step
+ * 4. check loss
+ *   4-0. save checkpoint for recovery/resume training
+ *   4-1. no more traning -> go to 5
+ *   4-2. more training -> go to 2
+ * 5. save current state to inference model
+ * 6. inference with inference model
+
+// sample example
+// 0. create context
+om_context *ctx;
+om_create_context(&ctx);
+
+// 1. load model
+om_load_model_from_file(ctx, MODEL_PATH);
+// 1-1. (optional, TBD) configure training options
+
+om_train_compile(ctx);
+
+float training_input[BATCH_SIZE*INPUT_SIZE];
+float training_label[BATCH_SIZE*OUTPUT_SIZE];
+
+// main training loop
+for(int epoch=0; epoch < NUM_EPOCHS; epoch++) {
+  for(int step=0; step < NUM_BATCHES ; step++) {
+    // prepare this steps's intput/label
+    memcpy(training_input, train_input_data + THIS_BATCH_OFFSET, BATCH_SIZE*INPUT_SIZE);
+    memcpy(training_output, train_output_data + THIS_BATCH_OFFSET, BATCH_SIZE*OUTPUT_SIZE);
+
+    // 2. feed training input / output
+    om_train_set_input(ctx, 0 , training_input, BATCH_SIZE*INPUT_SIZE);
+    om_train_set_expected(ctx, 0 , training_input,BATCH_SIZE*INPUT_SIZE);
+    // 3. train a step
+    om_train_single_step(ctx);
+  }
+  // 4. check loss
+  float loss;
+  om_train_get_loss(ctx, 0, &loss);
+
```

Comment:
I'd like to keep onert-micro as simple as possible since a variant introduces more maintenance cost. IMHO, it is better to implement at platform(aifw) level. That is, while onert-micro provides basic feature like this api, AIFW can provide high level features by assembling them.